### PR TITLE
server: fix issue that vm guest os type is reset after updatetemplate

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

Steps to reproduce the issue
(1) change vm guest os type
(2) update vm template (guest os type is not changed)
(3) vm guest os type is reset to guest os type of template.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
lgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@shwstppr i don't think these errors are relevant, can you comment? cc @rhtyd

File: server/src/main/java/com/cloud/template/TemplateManagerImpl.java [hunk 1]
```diff
@@ -2063,6 +2063,7 @@ private VMTemplateVO updateTemplateOrIso(BaseUpdateTemplateOrIsoCmd cmd) {
             ex.addProxyObject(String.valueOf(id), "templateId");
             throw ex;
         }
+        long oldGuestOSId = template.getGuestOSId();
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/template/TemplateManagerImpl.java:2066]: moving before this line L#2096 `template = _tmpltDao.createForUpdate(id);` should be ok?

**Comment by shwstppr** [comment]
k8s smoke test failure is env related. It failed to get ISO in ready state cc @DaanHoogland

**Review by shwstppr — APPROVED** [review]
LGTM

**Pull request merged**